src/shogun/clustering/Hierarchical.cpp

```diff
@@ -212,8 +216,86 @@ SGMatrix<int32_t> CHierarchical::get_cluster_pairs()
 	return SGMatrix<int32_t>(pairs,2,merges, false);
 }
 
-
 void CHierarchical::store_model_features()
 {
-	/* TODO. Currently does nothing since apply methods are not implemented. */
+	CDenseFeatures<float64_t>* rhs =
+	    dynamic_cast<CDenseFeatures<float64_t>*>(distance->get_rhs());
+
+	int32_t num_vectors = rhs->get_num_vectors();
+	int32_t num_features = rhs->get_num_features();
+
+	int32_t* n_cluster_samples = SG_MALLOC(int32_t, num_vectors + merges);
```

Comment:
using SGVector can make it clearer since you don't need `SG_MALLOC` / `SG_FREE`

Reply:
Made the change